### DISCUSSION TITLE
已下载视频加载速度慢

### DISCUSSION THREAD

**Discussion by lyppiao** [ywmoyue/biliuwp-lite#745] [General]
电脑配置比较低，在已下载大量视频（几十G）的情况下，加载100%的速度很慢，是哪方面的原因呢？
CPU、硬盘、内存、显卡，升级哪个硬件能显著提升速度？硬盘剩余空间极少，会不会也有影响？
还是说软件本身受限，硬件没有影响？

**Comment by lyppiao** [comment]
另外这个好像后台切换其他软件后，会停止加载，要回到软件前台后，才会继续加载

**Comment by ywmoyue** [comment]
这个是加载下载视频这段代码的时间复杂度高的问题，后面我定位一下看看能不能优化吧

**Comment by ywmoyue** [comment]
@lyppiao 试试4.7.4设置中启用这个功能后加载已下载视频速度有没有提升
![image](https://github.com/user-attachments/assets/3815581e-b9e9-48e8-8f69-e7e57d8da659)

**Reply by lyppiao** [reply]
> @lyppiao尝试4.7.4设置中启用该功能后加载已下载视频速度有没有提升 ![图像](https://private-user-images.githubusercontent.com/28388300/372378696-3815581e-b9e9-48e8-8f69-e7e57d8da659.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc4NzE2NjUsIm5iZiI6MTcyNzg3MTM2NSwicGF0aCI6Ii8yODM4ODMwMC8zNzIzNzg2OTYtMzgxNTU4MWUtYjllOS00OGU4LThmNjktZTdlNTdkOGRhNjU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAyVDEyMTYwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3ZmIwYTAwODE3YzQ0Mjk2NDcxYWY5ZjIwZThhN2EzZDgyNTIxMjgxODc5ZDI3ZDc2YjBlMzZhNzE2ODVjYmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QYThySDOYjVgEyAAVEjYxbvNSPnQtOO1zNLyLTy4-Ws)

挺好，352个视频合计33G，加载时间从42秒提升到了14秒。
另外，后台切换到其他软件后，仍然会停止加载，这个机制不变。这个无所谓，不影响使用的，索引快了就没有这个需求了。

然后现在的索引机制是，每次，关闭下载界面后重新打开，都要重新加载一遍。不过因为加载很快，也没有什么影响。
我想问一下这个加载很吃性能吗？如果性能占用低，能不能做成开启软件时，自动在后台开始索引，且后续开关下载界面，无需重新加载。

**Comment by ywmoyue** [comment]
@lyppiao 可以试试4.7.6，现在每次打开下载页都不会重新加载了

**Reply by lyppiao** [reply]
> @lyppiao可以试试4.7.6，现在每次打开下载页面都不会重新加载了

概率性卡9%进度后不加载

**Reply by lyppiao** [reply]
> @lyppiao 可以试试4.7.6，现在每次打开下载页都不会重新加载了

下载功能也同时失效了。之前开索引下载的是不能播放的。
现在关闭索引后，点下载以后不会进行真实下载。且搜索功能同时存在bug，搜完以后下载页就一直只有搜索出来的文件，无法退出到之前显示全部文件的界面。只有关闭整个下载页，重新打开，再走一轮加载才行。

**Reply by ywmoyue** [reply]
> 下载功能也同时失效了。之前开索引下载的是不能播放的。
> 现在关闭索引后，点下载以后不会进行真实下载。且搜索功能同时存在bug，搜完以后下载页就一直只有搜索出来的文件，无法退出到之前显示全部文件的界面。只有关闭整个下载页，重新打开，再走一轮加载才行。
> 

好的，感谢反馈

**Reply by lyppiao** [reply]
补充，当前4.7.7.801，对下载功能的测试。
同一个视频，第一次下载为假下载，表面显示成功通知，但不会进行真实下载。第二次下载，才会进行真实下载。
比如第一次连续下载了三个视频，全部为假下载。第二轮下载的时候，三个全部成功。

**Reply by lyppiao** [reply]
> @lyppiao 可以试试4.7.6，现在每次打开下载页都不会重新加载了

4.7.7.801，最新发现，我怀疑不是假下载，而是第一次搜不到。
我在一次下载之后，搜不到视频。然后关闭软件，重启，就能搜到了。
这样来说，之前所谓的第二次真下载，应该也只是刷新的一种方式，跟重启刷新差不多。

**Reply by lyppiao** [reply]
应该就是这个东西了，大概率是排查到了。
关闭索引时，加载的很慢，且当前第一次下载的视频不会被识别到。所以需要 第二次下载 或 重启软件 来刷新识别。
开启索引时，现在是从零加载，然后一次性展示，所以第一次下载就能识别到，已经正常了。
之前旧版本开索引不正常，可能也是一边加载一边展示的原因，导致没识别到新视频。

另外就是搜索到指定视频后，无法退出返回之前的全部视频界面，只能点刷新，重新读取一遍全部已下载视频，希望可以优化下。

**Reply by lyppiao** [reply]
4.7.7.801，出现多次下载均显示成功后，实际上没有进行下载的情况。
我怀疑会不会跟视频源也有关系，这个视频一直没办法进行真实下载。BV1KbzvYEEtw

**Reply by ywmoyue** [reply]
> 另外就是搜索到指定视频后，无法退出返回之前的全部视频界面，只能点刷新，重新读取一遍全部已下载视频，希望可以优化下。

这个可以清空搜索框再搜索就是返回全部视频界面，是希望加个按钮一键清空搜索结果吗

> 这个视频一直没办法进行真实下载。BV1KbzvYEEtw

测了下可以下载，不过下载选择列表这里确实有个bug，显示合集列表后重复显示了这个视频

**Reply by ywmoyue** [reply]
感觉可以优化下排序，
加一个排序方式选择根据下载时间排序和名称排序

**Reply by lyppiao** [reply]
> 这个可以清空搜索框再搜索就是返回全部视频界面，是希望加个按钮一键清空搜索结果吗？
> 
这个返回方式比较隐秘，你不说我都不知道还能这样。希望能额外加个按纽一键返回。
